miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultSampleService.java

```diff
@@ -325,12 +325,21 @@ private Sample save(Sample sample, boolean validateAliasUniqueness) throws IOExc
       // post-save field generation
       boolean needsUpdate = false;
       if (hasTemporaryName(sample)) {
-        created.setName(namingScheme.generateNameFor(created));
+        try {
+          created.setName(namingScheme.generateNameFor(created));
+        } catch (MisoNamingException e) {
```

Comment:
Does `MisoNamingException` serve a purpose anymore? Can it just be replaced with `ValidationError` in all cases? (Not for this PR. Just musing)

Reply:
We'd have to either move all of the naming scheme stuff into the service module, or the validation stuff into the core module/ The former probably makes more sense and I think would be ok. The naming schemes should only be used by the service and miso-web modules I think